src/components/Table/TableHeader.tsx

```diff
@@ -34,13 +35,14 @@ export interface ITableHeaderProps {
  *
  * - Renders drag & drop components
  */
-export const TableHeader = memo(function TableHeader({
-  headerGroups,
+export const TableHeader = function TableHeader({
+  table,
```

Comment:
I had to change `headerGroups` to `table` because `headerGroups` don't update on row selection.

Reply:
Hey @il3ven , I tried using `headerGroups` and keeping the memo on `TableHeader` and `TableBody`, and it's working fine for me. Could you please check it and let me know if I'm missing anything? Thank you!

Reply:
It is working right now with `headerGroups` and `memo`. It doesn't make sense at the moment but it is working. We can keep `memo` and `headerGroups`.

Reply:
Great! So, can you make the changes, as now it is working with `headerGroups` and `memo`.

Reply:
> So, can you make the changes, as now it is working with headerGroups and memo.

@iamanishroy Done!